### DISCUSSION TITLE
S3 client (java) returns question marks instead of filename

### DISCUSSION THREAD

**Discussion by nur-akazikulov** [minio/minio#16829] [Q&A] [answered]
## NOTE
I'm trying to get the metadata from the file that is saved in minio (S3). When I go and check it in the browser console, it shows correctly the filename in the metadata section: "X-Amz-Meta-Original-File-Name: name.blah". But the problem is when I'm doing it using java client, it returns "???.ext" question marks instead of the original filename when there's cyrillic symbols in the filename (even tho it shows normally in the console). Is there any setting I need to set in order to get normal string returned or it's just a bug?

## Expected Behavior
Return normal filename which includes cyrillic and maybe other language symbols which are displayable by utf-8.

## Current Behavior
Return "???.extension"

## Steps to Reproduce (for bugs)

```java
public class Test {
    private final AmazonS3 s3Client;

    public ObjectMetadata getObjectMetaData(GetObjectMetadataRequest metaDataRequest) {
        return s3Client.getObjectMetadata(metaDataRequest);
    }

...
metadata.getUserMetadata().get(filename_key);
...
}
``` 
## Your Environment

* Version used (`minio --version`): RELEASE.2022-05-08T23-50-31Z
* Server setup and configuration: 
* Operating System and version (`uname -a`): Linux minio (this is what `uname -a` returns, it's run in the kubernetes)

**Comment by klauspost** [comment, marked as answer]
Sounds like your client doesn't support the encoding sent. There is no fixed standard for this - I believe we just roundtrip the data as it was sent, but I am not 100 percent on that. UTF-8 can be roundtripped safely.

I suggest you base64 encode or otherwise escape your payload.
